README.md

```diff
@@ -42,126 +54,252 @@ SELECT
-    SELECT col1 \ 
-    FROM [10 * cos(i * ((pi * 4) / 90)) for i in range(80)] \
-    TO csv" | \ 
-sed 1d | \ 
-feedgnuplot --lines --points --exit
+cat file.yaml | yq -c | spyql "SELECT json->a_field FROM json"
+```
+```sh
+cat file.xml | xq -c | spyql "SELECT json->a_field FROM json"
+```
+```sh
+cat file.toml | tomlq -c | spyql "SELECT json->a_field FROM json"
 ```
+ 
 
 
+### Kafka to PostegreSQL pipeline
 
-## TODO list before official release
+Read data from a kafka topic and write to postgres.
 
-- [ ] AS table to identify relation (e.g. when writting to SQL)
-- [ ] reading NULL/None + tests
-- [ ] writing NULL/None + tests
-- [ ] limit the scope of variables when evaluating (https://stackoverflow.com/questions/2220699/whats-the-difference-between-eval-exec-and-compile)
-- [ ] RE match function
-- [ ] `like` function
-- [ ] go through SQL opertators/keywords, create functions and include them in tests
-- [ ] options for CSVs and alike (in)
-- [ ] options for CSVs and alike (out)
-- [ ] add option to disable buffering (-u) by calling flush every write
-- [ ] infer data types on csv
-- [ ] friendly exception reporting and errors messages
-- [ ] clean-up repo: https://docs.python-guide.org/writing/structure/
-- [ ] installation package, e.g.: https://github.com/glamp/bashplotlib/blob/master/setup.py
-- [ ] argument parsing: https://stackabuse.com/command-line-arguments-in-python/
-   --unbuffered (-u)
-   --sample-size (-s? CSV, JSON, ..), 
-   --batch-size (SQL)
-   --quiet -q, --verbose -v
-- [ ] command line help   
-- [ ] 90% code coverage
+NOTE: `sql  options not yet implemented`
 
-#### Tentative
-- [ ] write/load to/from py (write lists, or csv with header with python data types)
-- [ ] create a variable for each JSON field: requires reading a sample, parsing it, store parsed objs, and collecting all fields to a Set
-OR
-create a syntax to simplify reading json: e.g. `json$vehicle_id, json$data$start_ts` VS. `json['vehicle_id'], json['data']['start_ts']`
-- [ ] ORDER BY (in memory for now)
-- [ ] add badges to Readme.md: https://dev.to/ananto30/how-to-add-some-badges-in-your-git-readme-github-gitlab-etc-3ne9
-- [ ] Better handling of dates constants. e.g. instead of `datetime.datetime(2020,11,10,19,55,03)` use `$2020-11-10 19:55:04$` or `to_datetime('2020-11-10 19:55:04')`
+```sh
+kafkacat -b the.broker.com -t the.topic | 
+spyql "
+	SELECT 
+		json->customer->id AS id,
+		json->customer->name AS name
+	FROM json
+	TO sql('customer_table_name', chunk_size=1)
+" |
+psql -U an_user_name -h a.host.com a_database_name
+```
 
+### Sub-queries (piping)
 
-#### Next
+```sh
+cat a_file.json |
+spyql "
+	SELECT ' '.join[json->first_name, json->midle_name, json->last_name] AS full_name
+	FROM json 
+	TO spy" |
+spyql "SELECT full_name, full_name.upper() FROM spy"
+```
 
-- [ ] `GROUP BY` and aggregation functions (in memory for now)
-- [ ] `SELECT DISTINCT`
+### Queries over APIs
 
+```sh
+curl https://reqres.in/api/users?page=2 | 
```

Comment:
I didn't know about [reqres.in](reqres), I've used [httpbin.org](httpbin) in the past but this is also useful :clap: